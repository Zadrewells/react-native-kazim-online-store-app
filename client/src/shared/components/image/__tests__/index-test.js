import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {
	Image
} from 'src/shared/components';

describe('<Image />', () => {
	const getWrapper = props => renderer.create(
		<Image
			{
			...props
			}
		/>
	).toJSON();

	const modelI = {
		source: 'https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
		thumbnailSource: 'https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=50&w=50',
		width: 500,
		height: 500
	};

	it('renders correctly', () => {
		const tree = getWrapper(modelI);
		expect(tree).toMatchSnapshot();
	});
});
