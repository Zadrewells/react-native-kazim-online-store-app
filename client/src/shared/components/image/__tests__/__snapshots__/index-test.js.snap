// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly 1`] = `
<View
  height={500}
  source="https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
  style={
    Array [
      Object {
        "backgroundColor": "#d0d0db",
        "height": 500,
        "width": 500,
      },
    ]
  }
  thumbnailSource="https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=50&w=50"
  width={500}
>
  <Image
    blurRadius={0.5}
    fadeDuration={0}
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=50&w=50",
      }
    }
    style={
      Object {
        "height": 500,
        "opacity": 0,
        "width": 500,
      }
    }
  />
  <Image
    fadeDuration={0}
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "https://images.pexels.com/photos/1166990/pexels-photo-1166990.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
      }
    }
    style={
      Object {
        "bottom": 0,
        "height": 500,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 500,
      }
    }
  />
</View>
`;
